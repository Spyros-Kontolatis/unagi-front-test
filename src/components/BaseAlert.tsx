import React from 'react';
import {
  AlertWrapper,
  AlertHideButton,
  ErrorAlertWrapper,
  SuccessAlertWrapper,
  WarningAlertWrapper,
} from './BaseAlert.style';
import type { ReactNode } from 'react';
import type { BaseAlert } from '../types';

export default ({ message, type, close }: BaseAlert) => {
  const renderAlertByType = (children: ReactNode) => {
    switch (type) {
      case 'error':
        return <ErrorAlertWrapper>{children}</ErrorAlertWrapper>;
      case 'success':
        return <SuccessAlertWrapper>{children}</SuccessAlertWrapper>;
      case 'warning':
        return <WarningAlertWrapper>{children}</WarningAlertWrapper>;
      default:
        return <AlertWrapper>{children}</AlertWrapper>;
    }
  };
  return renderAlertByType(
    <>
      {!!close && (
        <AlertHideButton data-testid="alertClose" onClick={() => close()}>
          &times;
        </AlertHideButton>
      )}
      {message}
    </>,
  );
};
