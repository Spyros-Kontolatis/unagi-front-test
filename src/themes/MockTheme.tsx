import React, { ReactNode } from 'react';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import customTheme from './customTheme';

function MockTheme({ children }: { children: ReactNode }) {
  const theme = createTheme(customTheme);
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

export default MockTheme;
