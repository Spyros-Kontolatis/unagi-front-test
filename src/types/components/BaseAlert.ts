export type BaseAlertType = 'info' | 'success' | 'error' | 'warning';
export interface BaseAlert {
  message: string;
  type?: BaseAlertType;
  close?: () => void;
}
