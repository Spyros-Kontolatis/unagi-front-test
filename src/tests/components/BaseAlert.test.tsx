import BaseAlert from '../../components/BaseAlert';

import MockTheme from '../../themes/MockTheme';
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';

describe('BaseAlert', () => {
  test('renders the component', () => {
    const { baseElement } = render(
      <MockTheme>
        <BaseAlert message="Info" />
      </MockTheme>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  test('renders the component as success', () => {
    const { baseElement } = render(
      <MockTheme>
        <BaseAlert message="Success" type="success" />,
      </MockTheme>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  test('renders the component as error', () => {
    const { baseElement } = render(
      <MockTheme>
        <BaseAlert message="Error" type="error" />,
      </MockTheme>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  test('renders the component as warning', () => {
    const { baseElement } = render(
      <MockTheme>
        <BaseAlert message="Warning" type="warning" />,
      </MockTheme>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  test('calls alert close', () => {
    const close = jest.fn();
    render(
      <MockTheme>
        <BaseAlert message="Info" close={close} />
      </MockTheme>,
    );

    const closeButton = screen.getByTestId('alertClose');
    expect(closeButton).toBeTruthy();
    fireEvent.click(closeButton);
    expect(close).toHaveBeenCalledTimes(1);
  });
});
