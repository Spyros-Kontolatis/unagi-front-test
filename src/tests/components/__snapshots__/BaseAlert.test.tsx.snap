// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseAlert renders the component 1`] = `
<body>
  <div>
    <div
      class="css-o7gs71"
    >
      Info
    </div>
  </div>
</body>
`;

exports[`BaseAlert renders the component as error 1`] = `
<body>
  <div>
    <div
      class="css-1ozbvuf"
    >
      Error
    </div>
    ,
  </div>
</body>
`;

exports[`BaseAlert renders the component as success 1`] = `
<body>
  <div>
    <div
      class="css-xglbxr"
    >
      Success
    </div>
    ,
  </div>
</body>
`;

exports[`BaseAlert renders the component as warning 1`] = `
<body>
  <div>
    <div
      class="css-129cgic"
    >
      Warning
    </div>
    ,
  </div>
</body>
`;
